model/task_queue_service.go

```diff
@@ -367,6 +368,28 @@ func (d *basicCachedDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueItem {
 			}
 
 			if !dependenciesMet {
+				satisfiable, err := nextTaskFromDB.DependencySatisfiable(dependencyCaches)
+				grip.Warning(message.WrapError(err, message.Fields{
+					"dispatcher": SchedulableUnitDispatcher,
+					"function":   "FindNextTask",
+					"operation":  "satisfiable",
+					"message":    "error checking dependencies for task",
+					"outcome":    "skip and continue",
+					"task":       item.Id,
+					"distro_id":  d.distroID,
+				}))
+				if err == nil && !satisfiable {
+					grip.Warning(message.WrapError(nextTaskFromDB.DeactivateTask(evergreen.DefaultTaskActivator+".dispatcher"),
```

Comment:
You've added this in both the revised and with-dependencies dispatchers. We know the former works, since we're running it in production, but we're less sure about the latter. I'm concerned about a risky change to the revised dispatcher. (Feature flagging new scheduler features has been very useful, but we wouldn't be doing that here.)

Could we as an alternative deploy this without deactivating these tasks, only logging them? Or do this only in the tunable dispatcher? I think it'd be worth validating `DependencySatisfiable` with either careful unit testing or with in-production testing, since this is a potentially dangerous or expensive change to the scheduler.

Reply:
I reverted the change to revised, so that this change only happens for tunable/with dependencies. 

I think it's actually not that expensive, because we cache the list of possible dependencies between the dependencies met (which we're already doing) and the satisfiability check (which should arguably really be part of the same, but it's fine.) 

I think the dependencyMet check works and ensures that we don't dispatch tasks with unsatisfied dependencies. the dependenciesSatisfiable, just ensures that we don't *leave* a blocked task in the queue if its impossible for the task to be activated. I think the task-lifecycle machinery also helps a bunch as a backstop to activate tasks which should be activated.